vpr/src/base/vpr_types.h

```diff
@@ -208,6 +212,42 @@ class t_ext_pin_util_targets {
     std::map<std::string, t_ext_pin_util> overrides_;
 };
 
+class t_allow_unrelated_clustering {
+  public:
+    t_allow_unrelated_clustering() = default;
+    t_allow_unrelated_clustering(enum e_unrel_clust_stat status, enum e_unrel_clust_mode mode);
+    /**
+     * @brief Returns the unrelated clustering status for the specified block type
+     */
+    enum e_unrel_clust_stat get_block_status(std::string block_type_name) const;
+    /**
+     * @brief Returns the unrelated clustering mode for the specified block type
+     */
+    enum e_unrel_clust_mode get_block_mode(std::string block_type_name) const;
+
+  public:
+    /**
+     * @brief Sets the unrelated clustering status for the specified block type
+     * 
+     * The unrelated clustering status is represented as a pair of booleans. 
+     * The first boolean incidcates whether unrelated clustering is turned on or off. 
+     * The second variable indicates whether the status of the variable is provided
+     * by the user(bool = false) or user let it to be automatically determined by VPR(bool = ture). 
+     * If provided by the user, the status cannot be overriden later, otherwise
+     * it can be adjusted by VPR throughout the flow as necessary.
+     */
+    void set_block_status(std::string block_type_name, std::pair<enum e_unrel_clust_stat, enum e_unrel_clust_mode> status);
+    /**
+     * @brief Sets the default value for unrelated clustering status
+     * 
+     */
+    void set_default_status(std::pair<enum e_unrel_clust_stat, enum e_unrel_clust_mode> status);
+
+  private:
+    std::pair<enum e_unrel_clust_stat, enum e_unrel_clust_mode> default_;
```

Comment:
Comment what these are: default value for all block types, and overrides by block_type name for specific types. Each says whether unrelated cluster is on or off, and if it was set to that state by the user or automatic vpr behaviour.